eclipse-scout-core/src/desktop/hybrid/converter/HybridActionContextElementConverter.ts

```diff
@@ -0,0 +1,87 @@
+
+/**
+ * Instances of this class are used to convert a model element to a JSON representation and vice-versa. Each
+ * instance typically only handles a specific type of element. Because elements have no common structure, they are
+ * always accompanied by the widget that owns them (here represented by the corresponding {@link ModelAdapter}).
+ *
+ * How to add a converter:
+ * - Create a subclass of this abstract class.
+ * - Implement the abstract methods {@link #jsonToElement} and {@link #elementToJson}. Return `null` if the
+ *   converter cannot handle the given arguments (in which case the task is delegated to the next converter).
+ * - Register the class using the static {@link register} method.
+ */
+export abstract class HybridActionContextElementConverter {
+
+  static DEFAULT_ORDER = 5000.0;
+  protected static _registry: HybridActionContextElementConverterRegistration[] = [];
+
+  static register(converter: HybridActionContextElementConverter | Constructor<HybridActionContextElementConverter>, order = HybridActionContextElementConverter.DEFAULT_ORDER) {
+    let registration: HybridActionContextElementConverterRegistration = {
+      order: order,
+      converter: converter instanceof HybridActionContextElementConverter ? converter : null,
+      converterConstructor: converter instanceof HybridActionContextElementConverter ? null : converter
+    };
+    arrays.insertSorted(HybridActionContextElementConverter._registry, registration, (r1, r2) => r1.order - r2.order);
+  }
+
+  static all(): HybridActionContextElementConverter[] {
+    return HybridActionContextElementConverter._registry.map(registration => {
+      registration.converter = registration.converter || scout.create(registration.converterConstructor);
+      return registration.converter;
+    });
+  }
+
+  static convertFromJson(adapter: ModelAdapter, jsonElement: any): HybridActionContextElement {
+    let converters = HybridActionContextElementConverter.all();
+    for (let i = 0; i < converters.length; i++) {
+      let element = converters[i].jsonToElement(adapter, jsonElement);
+      if (element) {
+        return element;
+      }
+    }
+    return null;
+  }
+
+  static convertToJson(adapter: ModelAdapter, element: any): JsonHybridActionContextElement {
+    let converters = HybridActionContextElementConverter.all();
+    for (let i = 0; i < converters.length; i++) {
+      let jsonElement = converters[i].elementToJson(adapter, element);
+      if (jsonElement) {
+        return jsonElement;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * @param adapter owner widget (e.g. TreeAdapter)
+   * @param jsonElement JSON representation of the element (e.g. String)
+   * @return model representation of the element (e.g. TreeNode), or `null` if this instance cannot handle the
+   * given arguments (in which case the task is delegated to the next converter)
+   */
+  abstract jsonToElement(adapter: ModelAdapter, jsonElement: any): any;
```

Comment:
I tried this, but it did not make really make the implementation easier. In fact, the code is longer, because it needs to overwrite 4 methods instead of 2.

Reply:
Ok, I'm convinced. The implementation was adjusted to match the Java equivalent.